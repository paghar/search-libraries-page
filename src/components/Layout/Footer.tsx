import Facebook from "@/components/misc/svg/Facebook";
import Twitter from "@/components/misc/svg/Twitter";
import Instagram from "@/components/misc/svg/Instagram";
import Linkedin from "@/components/misc/svg/Linkedin";
import GitHub from "@/components/misc/svg/GitHub";
import Email from "@/components/misc/svg/Email";
import Phone from "@/components/misc/svg/Phone";
import {footerItems} from "@/utils/constants/footer";
import {IFooterItem} from "@/utils/interface/misc";

const Footer = () => {

  const items = footerItems?.map((item:IFooterItem)=>{        
    return <li 
      className="p-3 hover:font-semibold" 
      key={item.value} >
      {item.text}
    </li>; 
  });

  return(
    <footer className="w-full py-6 px-4 bg-pink-700"> 
      <div className="mx-auto max-w-7xl flex flex-col items-center text-white">
        <div className="w-full flex flex-col flex-wrap justify-between items-center">
          {/* :SOCIALMEDIA */}
          <div className="py-4 px-2 flex">
            <Facebook className="w-6 h-6 mr-1"/>
            <Twitter className="w-6 h-6 mr-1"/>
            <Instagram className="w-6 h-6 mr-1"/>
            <Linkedin className="w-6 h-6 mr-1"/>
            <GitHub className="w-6 h-6 mr-1"/>          
          </div>
          {/* :NAVIGATION */}
          <nav className="py-4 px-2 flex flex-wrap justify-center list-none">
            {items}           
          </nav>
          {/* :CONTACT */}
          <div className="px-2 flex flex-col">
            {/* Email */}
            <p className="flex text-xs font-medium tracking-wide">
              <Email className="mr-1 h-5 w-5"/>            
              <a href="#email">fatemeh.paghar@outlook.com</a>
            </p>
            {/* Phone */}
            <p className="flex text-xs  font-bold">
              <Phone className="mr-1 h-5 w-5" />              
              <span>176-44544529</span>
            </p>
          </div>
        </div>
        {/* :COPYRIGHT */}
        <p className="pt-10 text-sm text-center">&copy;2024, Fancy Tailwind,React,Next.js,TypeScript All Rights Reserved.</p>
      </div>
    </footer>
  );
};

export default Footer;