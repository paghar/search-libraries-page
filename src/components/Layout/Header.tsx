import Search from "@/components/misc/svg/Search";
import Button from "@/components/misc/Button";
import Textbox from "@/components/misc/Textbox";
import {SearchAction} from "@/utils/enum/search";

interface IProps{   
  searchText:string
  dispatch:any
  searchClick:()=>void
}

const Header = ({searchText,dispatch,searchClick}:IProps) => {

  const borderRadiusStyle = {
    borderBottomLeftRadius: "50% 20%",
    borderBottomRightRadius: "50% 20%",
  };

  return(    
    <div className="overflow-hidden relative flex items-center min-h-32 w-full">  
      <div style={borderRadiusStyle} className="absolute bg-pink-700 w-full h-1/2 z-0 top-0"/>
      <div className="absolute inset-x-auto w-full z-10">
        <div className="sm:w-full md:w-3/5 lg:w-3/5 xl:w-2/5 2xl:w-2/5 mx-auto shadow-md rounded-md p-4 bg-white">           
          <form className="flex items-center justify-center 2xl:flex-nowrap xl:flex-nowrap lg:flex-nowrap md:flex-wrap sm:flex-wrap "> 
            <div className="relative 2xl:w-5/6 xl:w-5/6 lg:w-5/6 md:w-4/6 md:mb-2 sm:w-full sm:mb-2">                
              <div className="flex absolute inset-y-0 left-0 items-center pl-3 pointer-events-none">
                <Search className={`"w-5 h-5 text-gray-500 dark:text-gray-400`}/>            
              </div>
              <Textbox
                id="search"                
                value={searchText}
                type="text"
                placeholder="Search ..."
                onChange={(e) => dispatch({type: SearchAction.SetSearchText,payload:e.target.value})} 
              />      
            </div>
            <Button
              id="search"
              type="button"             
              onClick={searchClick}
              text="Search"
              className="w-28 "
            >
              <Search className={`mr-2 -ml-1 w-5 h-5`}/>
            </Button>                     
          </form>         
        </div>
      </div>
    </div>
  );
};

export default Header;