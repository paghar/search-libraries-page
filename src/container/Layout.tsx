import Layout from "@/components/Layout/Layout";
import {useSearch, useSearchDispatch} from "@/context/searchInfo";
import {sort} from "@/hooks/sort";
import {fetchData} from "@/hooks/useSearch";
import {SearchAction} from "@/utils/enum/search";
import {InitialStateProps} from "@/utils/interface/search";

const LayoutContainer = ({children}: any) => {

  const searchInfos:InitialStateProps = useSearch();
  const dispatch = useSearchDispatch();  

  const perPage = 5;
 
  const searchClick = () => {
    const key="f276c74b4b974df2432cede6284ff105"; 
    dispatch({type: SearchAction.setPageNumber,payload:1});  
    fetchData(searchInfos.searchText,sort(searchInfos.sort),key,dispatch,1,perPage); 
  };
  
  return (
    <Layout searchText={searchInfos.searchText} dispatch={dispatch} searchClick={searchClick} >
      {children}
    </Layout>
  );
};

export default LayoutContainer;
