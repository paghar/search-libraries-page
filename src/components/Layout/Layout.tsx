import Footer from "./Footer";
import Header from "./Header";

interface IProps{ 
  children:any 
  searchText:string
  dispatch:any
  searchClick:()=>void
}

const Layout = ({searchText,children,dispatch,searchClick}: IProps) => {
  return (
    <>
      <Header searchText={searchText} dispatch={dispatch} searchClick={searchClick}/>
      {children}
      <Footer/>
    </>
  );
};

export default Layout;