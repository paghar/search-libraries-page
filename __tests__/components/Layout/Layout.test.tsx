import React from "react";
import {render} from "@testing-library/react";
import Layout from "../../../src/components/Layout/Layout";

const searchText = "Test search";
const dispatch = jest.fn();
const searchClick = jest.fn();
  
describe("Layout Component", () => {

  const {container} = render(   
    <Layout searchText={searchText} dispatch={dispatch} searchClick={searchClick}>
      <div>Test children</div>
    </Layout>
  );
  
  it("Renders Snapshot Layout", () => {    
    expect(container).toMatchSnapshot();
  });
  
});

describe("Layout Component", () => {
  test("renders children, Header, and Footer components with provided props", () => {
    
    const {getByText} = render(
      <Layout searchText={searchText} dispatch={dispatch} searchClick={searchClick}>
        <div>Test children</div>
      </Layout>
    );
    
    // Ensure children are rendered
    const testChild = getByText("Test children");
    expect(testChild).toBeInTheDocument();
  });
});
