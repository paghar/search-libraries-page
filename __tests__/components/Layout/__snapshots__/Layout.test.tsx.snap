// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout Component Renders Snapshot Layout 1`] = `
<div>
  <div
    class="overflow-hidden relative flex items-center min-h-32 w-full"
  >
    <div
      class="absolute bg-pink-700 w-full h-1/2 z-0 top-0"
      style="border-bottom-left-radius: 50% 20%; border-bottom-right-radius: 50% 20%;"
    />
    <div
      class="absolute inset-x-auto w-full z-10"
    >
      <div
        class="sm:w-full md:w-3/5 lg:w-3/5 xl:w-2/5 2xl:w-2/5 mx-auto shadow-md rounded-md p-4 bg-white"
      >
        <form
          class="flex items-center justify-center 2xl:flex-nowrap xl:flex-nowrap lg:flex-nowrap md:flex-wrap sm:flex-wrap "
        >
          <div
            class="relative 2xl:w-5/6 xl:w-5/6 lg:w-5/6 md:w-4/6 md:mb-2 sm:w-full sm:mb-2"
          >
            <div
              class="flex absolute inset-y-0 left-0 items-center pl-3 pointer-events-none"
            >
              <svg
                class=""w-5 h-5 text-gray-500 dark:text-gray-400"
                fill="none"
                stroke="currentColor"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M21 21l-6-6m2-5a7 7 0 11-14 0 7 7 0 0114 0z"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                />
              </svg>
            </div>
            <input
              class="   bg-gray-50 border border-gray-300 text-gray-900 text-sm rounded-lg   focus:ring-pink-500 focus:border-pink-500 block w-full pl-10 p-2.5    dark:bg-gray-700 dark:border-gray-600 dark:placeholder-gray-400    dark:text-white dark:focus:ring-pink-500 dark:focus:border-pink-500   "
              id="search"
              placeholder="Search ..."
              type="text"
              value="Test search"
            />
          </div>
          <button
            class="w-28 
        inline-flex items-center 
        py-2.5 px-3 ml-2 text-sm 
        font-medium text-white
        bg-pink-700 rounded-lg border border-pink-700
        hover:bg-pink-800 focus:ring-4 
        focus:outline-none focus:ring-pink-300
        dark:bg-pink-600 dark:hover:bg-pink-700 dark:focus:ring-pink-800
      "
            id="search"
            type="button"
          >
            <svg
              class="mr-2 -ml-1 w-5 h-5"
              fill="none"
              stroke="currentColor"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M21 21l-6-6m2-5a7 7 0 11-14 0 7 7 0 0114 0z"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
            </svg>
            Search
          </button>
        </form>
      </div>
    </div>
  </div>
  <div>
    Test children
  </div>
  <footer
    class="w-full py-6 px-4 bg-pink-700"
  >
    <div
      class="mx-auto max-w-7xl flex flex-col items-center text-white"
    >
      <div
        class="w-full flex flex-col flex-wrap justify-between items-center"
      >
        <div
          class="py-4 px-2 flex"
        >
          <svg
            class="w-6 h-6 mr-1"
            data-testid="facebook-icon"
            fill="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
          >
            <path
              d="M18 2h-3a5 5 0 00-5 5v3H7v4h3v8h4v-8h3l1-4h-4V7a1 1 0 011-1h3z"
            />
          </svg>
          <svg
            class="w-6 h-6 mr-1"
            data-testid="twitter-icon"
            fill="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
          >
            <path
              d="M23 3a10.9 10.9 0 01-3.14 1.53 4.48 4.48 0 00-7.86 3v1A10.66 10.66 0 013 4s-4 9 5 13a11.64 11.64 0 01-7 2c9 5 20 0 20-11.5a4.5 4.5 0 00-.08-.83A7.72 7.72 0 0023 3z"
            />
          </svg>
          <svg
            class="w-6 h-6 mr-1"
            data-testid="instagram-icon"
            fill="none"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
          >
            <rect
              height="20"
              rx="5"
              ry="5"
              width="20"
              x="2"
              y="2"
            />
            <path
              d="M16 11.37A4 4 0 1112.63 8 4 4 0 0116 11.37zm1.5-4.87h.01"
            />
          </svg>
          <svg
            class="w-6 h-6 mr-1"
            data-testid="linkedin-icon"
            fill="currentColor"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="0"
            viewBox="0 0 24 24"
          >
            <path
              d="M16 8a6 6 0 016 6v7h-4v-7a2 2 0 00-2-2 2 2 0 00-2 2v7h-4v-7a6 6 0 016-6zM2 9h4v12H2z"
              stroke="none"
            />
            <circle
              cx="4"
              cy="4"
              r="2"
              stroke="none"
            />
          </svg>
          <svg
            aria-hidden="true"
            class="w-6 h-6 mr-1"
            data-testid="github-icon"
            fill="currentColor"
            viewBox="0 0 24 24"
          >
            <path
              clip-rule="evenodd"
              d="M12 2C6.477 2 2 6.484 2 12.017c0 4.425 2.865 8.18 6.839 9.504.5.092.682-.217.682-.483 0-.237-.008-.868-.013-1.703-2.782.605-3.369-1.343-3.369-1.343-.454-1.158-1.11-1.466-1.11-1.466-.908-.62.069-.608.069-.608 1.003.07 1.531 1.032 1.531 1.032.892 1.53 2.341 1.088 2.91.832.092-.647.35-1.088.636-1.338-2.22-.253-4.555-1.113-4.555-4.951 0-1.093.39-1.988 1.029-2.688-.103-.253-.446-1.272.098-2.65 0 0 .84-.27 2.75 1.026A9.564 9.564 0 0112 6.844c.85.004 1.705.115 2.504.337 1.909-1.296 2.747-1.027 2.747-1.027.546 1.379.202 2.398.1 2.651.64.7 1.028 1.595 1.028 2.688 0 3.848-2.339 4.695-4.566 4.943.359.309.678.92.678 1.855 0 1.338-.012 2.419-.012 2.747 0 .268.18.58.688.482A10.019 10.019 0 0022 12.017C22 6.484 17.522 2 12 2z"
              fill-rule="evenodd"
            />
          </svg>
        </div>
        <nav
          class="py-4 px-2 flex flex-wrap justify-center list-none"
        >
          <li
            class="p-3 hover:font-semibold"
          >
            Team
          </li>
          <li
            class="p-3 hover:font-semibold"
          >
            Terms and Conditions
          </li>
          <li
            class="p-3 hover:font-semibold"
          >
            Privacy Policy
          </li>
          <li
            class="p-3 hover:font-semibold"
          >
            API
          </li>
        </nav>
        <div
          class="px-2 flex flex-col"
        >
          <p
            class="flex text-xs font-medium tracking-wide"
          >
            <svg
              class="mr-1 h-5 w-5"
              fill="currentColor"
              viewBox="0 0 20 20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M2.003 5.884L10 9.882l7.997-3.998A2 2 0 0016 4H4a2 2 0 00-1.997 1.884z"
              />
              <path
                d="M18 8.118l-8 4-8-4V14a2 2 0 002 2h12a2 2 0 002-2V8.118z"
              />
            </svg>
            <a
              href="#email"
            >
              fatemeh.paghar@outlook.com
            </a>
          </p>
          <p
            class="flex text-xs  font-bold"
          >
            <svg
              class="mr-1 h-5 w-5"
              fill="none"
              stroke="currentColor"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M3 5a2 2 0 012-2h3.28a1 1 0 01.948.684l1.498 4.493a1 1 0 01-.502 1.21l-2.257 1.13a11.042 11.042 0 005.516 5.516l1.13-2.257a1 1 0 011.21-.502l4.493 1.498a1 1 0 01.684.949V19a2 2 0 01-2 2h-1C9.716 21 3 14.284 3 6V5z"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
            </svg>
            <span>
              176-44544529
            </span>
          </p>
        </div>
      </div>
      <p
        class="pt-10 text-sm text-center"
      >
        Â©2024, Fancy Tailwind,React,Next.js,TypeScript All Rights Reserved.
      </p>
    </div>
  </footer>
</div>
`;
